File: model-client/src/main/java/org/modelix/model/api/ITransaction.java
package org.modelix.model.api;

import java.util.stream.LongStream;

public interface ITransaction {
  IBranch getBranch();
  ITree getTree();

  boolean containsNode(long nodeId);
  IConcept getConcept(long nodeId);
  long getParent(long nodeId);
  String getRole(long nodeId);
  String getProperty(long nodeId, String role);
  INodeReference getReferenceTarget(long sourceId, String role);
  LongStream getChildren(long parentId, String role);
  LongStream getAllChildren(long parentId);
}

File: model-client/src/main/kotlin/org/modelix/model/api/ITransaction.kt
package org.modelix.model.api

import java.util.stream.LongStream

interface ITransaction {
    val branch: IBranch?
    var tree: ITree?
    fun containsNode(nodeId: Long): Boolean
    fun getConcept(nodeId: Long): IConcept?
    fun getParent(nodeId: Long): Long
    fun getRole(nodeId: Long): String?
    fun getProperty(nodeId: Long, role: String?): String?
    fun getReferenceTarget(sourceId: Long, role: String?): INodeReference?
    fun getChildren(parentId: Long, role: String?): LongStream?
    fun getAllChildren(parentId: Long): LongStream?
}